airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -501,83 +561,41 @@ def create_pod_request_obj(self) -> k8s.V1Pod:
         if self.do_xcom_push:
             self.log.debug("Adding xcom sidecar to task %s", self.task_id)
             pod = xcom_sidecar.add_xcom_sidecar(pod)
-        return pod
 
-    def create_new_pod_for_operator(self, labels, launcher) -> Tuple[State, k8s.V1Pod, Optional[str]]:
-        """
-        Creates a new pod and monitors for duration of task
-
-        :param labels: labels used to track pod
-        :param launcher: pod launcher that will manage launching and monitoring pods
-        :return:
-        """
-        self.log.debug(
-            "Adding KubernetesPodOperator labels to pod before launch for task %s", self.task_id
-        )
+        labels = self._create_labels_for_pod(context)
+        self.log.info("creating pod with labels %s and launcher %s", labels, self.launcher)
```

Comment:
```suggestion
        self.log.info("creating pod %s with labels: %s", pod.metadata.name, labels)
```

(untested) It'd be cool to have the pod name. Don't think logging the launcher is helpful.

Reply:
And we might want to move this down after we finish modifying the labels, right before we return.